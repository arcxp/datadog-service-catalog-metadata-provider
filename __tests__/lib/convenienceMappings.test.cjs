const YAML = require('yaml')
const core = require('@actions/core')
const subject = require('../../lib/convenienceMappings.cjs')

describe('lib/convenienceMappings.cjs', () => {
  test('#getConvenienceFieldValues() - full set', () => {
    const testInput = `
---
email: 'team-name-here@fakeemaildomainthatdoesntexist.com'
slack: 'https://fakeorg.slack.com/archives/A0000000000'
slack-support-channel: 'https://fakeorg.slack.com/archives/A0000000001'
repo: https://github.com/arcxp/datadog-service-catalog-metadata-provider
`
    // Full set of fields
    core.__setInputsObject(YAML.parse(testInput))
    const inputs = subject.getConvenienceFieldValues(core)
    expect(inputs).toMatchSnapshot()
    expect(inputs.email).toEqual(
      'team-name-here@fakeemaildomainthatdoesntexist.com',
    )
  })

  test('#getConvenienceFieldValues() - partial set', () => {
    const testInput = `
---
slack: 'https://fakeorg.slack.com/archives/A0000000000'
slack-support-channel: 'https://fakeorg.slack.com/archives/A0000000001'
repo: https://github.com/arcxp/datadog-service-catalog-metadata-provider
`
    // Full set of fields
    core.__setInputsObject(YAML.parse(testInput))
    const inputs = subject.getConvenienceFieldValues(core)
    expect(inputs).toMatchSnapshot()
    expect(inputs.email).toBeUndefined()
  })
})
