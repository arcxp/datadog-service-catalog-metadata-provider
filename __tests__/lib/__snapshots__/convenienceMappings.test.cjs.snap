// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/convenienceMappings.cjs #getConvenienceFieldValues() - full set 1`] = `
{
  "email": "team-name-here@fakeemaildomainthatdoesntexist.com",
  "repo": "https://github.com/arcxp/datadog-service-catalog-metadata-provider",
  "slack": "https://fakeorg.slack.com/archives/A0000000000",
  "slack-support-channel": "https://fakeorg.slack.com/archives/A0000000001",
}
`;

exports[`lib/convenienceMappings.cjs #getConvenienceFieldValues() - full set 2`] = `
{
  "repo": "https://github.com/arcxp/datadog-service-catalog-metadata-provider",
  "slack": "https://fakeorg.slack.com/archives/A0000000000",
  "slack-support-channel": "https://fakeorg.slack.com/archives/A0000000001",
}
`;

exports[`lib/convenienceMappings.cjs #getConvenienceFieldValues() - partial set 1`] = `
{
  "repo": "https://github.com/arcxp/datadog-service-catalog-metadata-provider",
  "slack": "https://fakeorg.slack.com/archives/A0000000000",
  "slack-support-channel": "https://fakeorg.slack.com/archives/A0000000001",
}
`;
